snuba/datasets/factory.py

```diff
@@ -53,12 +54,19 @@ def get_dataset(name: str) -> Dataset:
 
     try:
         dataset = DATASETS_IMPL[name] = dataset_factories[name]()
+        DATASETS_NAME_LOOKUP[dataset] = name
```

Comment:
It probably would be good to check that the dataset isn't already registered before writing it here, just to be safe — the input value enforces key uniqueness for the implementation mapping, but doesn't make any guarantees the other way.

Reply:
I don't think it's possible that the same dataset can be registered multiple times here. Since each `dataset` is an instance of a class it should map to a different key in `DATASETS_NAME_LOOKUP`.

Reply:
~Can't the same dataset constructor appear multiple times in `dataset_factories.values()` though?~

Nevermind, I guess that's your point — even if the constructor appears multiple times, the instances will be different. Oops. 😅